Shadowblade_main.lua

```diff
@@ -0,0 +1,330 @@
+		end
+	end
+	
+	if nAddBonus > 0 then
+		--decay before we add
+		core.DecayBonus(self)
+		core.nHarassBonus = core.nHarassBonus + nAddBonus
+	end
+end
+object.oncombateventOld = object.oncombatevent
+object.oncombatevent 	= object.oncombateventOverride
+------------------------------------------------------
+--            CustomHarassUtility Override          --
+-- Change Utility according to usable spells here   --
+------------------------------------------------------
+-- @param: IunitEntity hero
+-- @return: number
+local function CustomHarassUtilityFnOverride(hero)
+    local nUtil = 0
+	 
+    if skills.abilGargantuan:CanActivate() then
+        nUtil = nUtil + object.nGargantuanUp
+    end
+ 
+    if skills.abilFeint:CanActivate() then
+        nUtil = nUtil + object.nFeintUp
+    end
+ 
+    if skills.abilSoul:CanActivate() then
+        nUtil = nUtil + object.nSoulUp
+    end
+	
+
+
+    return nUtil
+end
+-- assign custom Harrass function to the behaviourLib object
+behaviorLib.CustomHarassUtility = CustomHarassUtilityFnOverride
+----------------------------------
+--	Shadowblade harass actions
+----------------------------------
+local function HarassHeroExecuteOverride(botBrain)
+	local bDebugEchos = false
+	
+	local unitTarget = behaviorLib.heroTarget
+	if unitTarget == nil or not unitTarget:IsValid() then
+		return false --can not execute, move on to the next behavior
+	end
+	
+	local unitSelf = core.unitSelf
+	local vecMyPosition = unitSelf:GetPosition()
+	
+	local vecTargetPosition = unitTarget:GetPosition()
+	local nTargetDistanceSq = Vector3.Distance2DSq(vecMyPosition, vecTargetPosition)
+	
+	local nLastHarassUtil = behaviorLib.lastHarassUtil
+	local bCanSee = core.CanSeeUnit(botBrain, unitTarget)	
+	
+	local bActionTaken = false
+	
+	local abilFeint = skills.abilFeint
+	local abilGargantuan = skills.abilGargantuan
+	local abilSoul = skills.abilSoul
+	local abilEssence = skills.abilEssence
+	
+--[[	-- bonus aggresion points for missing enemy HP  ---- Doesn't work
+    if bCanSee then
+		local nTargetHealth = unitTarget:GetHealthPercent()
+		nLastHarassUtil = nLastHarassUtil + (100 - nTargetHealth * 100)
+	end
+
+
+--]]
+
+	--Feint
+	if abilFeint:CanActivate() and bCanSee and unitSelf:GetMana() > 200 then
+		local nRange = abilFeint:GetRange()
+		if abilGargantuan:CanActivate() and nTargetDistanceSq < (nRange * nRange) then
+			bActionTaken = core.OrderAbilityEntity(botBrain, abilFeint, unitTarget)
+		end
+	end
+
+		--Gargantuan
+	if not bActionTaken and abilGargantuan:CanActivate() then
+		if nTargetDistanceSq < (300 * 300) then
+			bActionTaken = core.OrderAbility(botBrain, abilGargantuan)
+		--Soul is used only if Feint is on Cooldown, to be used in combo: Feint -> Gargantuan -> Soul
+			if abilFeint:CanActivate() then
+						bActionTaken = core.OrderAbilityEntity(botBrain, abilFeint, unitTarget)
+				elseif abilSoul:CanActivate() then
+						bActionTaken = core.OrderAbility(botBrain, abilSoul) 
+				end
+		end
+	end	
+		
```

Comment:
You may also want to incorporate a snippet for Soul (for ranged harass/farming capabilities) and for Essence Shift (grabbing any desirable attributes from the best target).